source/Irrlicht/COpenGLDriver.cpp

```diff
@@ -2427,21 +1989,27 @@ void COpenGLDriver::SAuxContext::flushState(GL_STATE_BITS stateBits)
 
                 if (STATE_NEQ(vertexInputParams.bindings[bnd]))//this if-statement also doesnt allow GlVertexArrayVertexBuffer be called multiple times for single binding
                 {
-                    extGlVertexArrayVertexBuffer(vao, bnd, nextState.vertexInputParams.bindings[bnd].buf, nextState.vertexInputParams.bindings[bnd].offset, hashVal.getStrideForBinding(bnd));
+                    extGlVertexArrayVertexBuffer(vao, bnd, nextState.vertexInputParams.bindings[bnd].buf->getOpenGLName(), nextState.vertexInputParams.bindings[bnd].offset, hashVal.getStrideForBinding(bnd));
                     UPDATE_STATE(vertexInputParams.bindings[bnd]);
                 }
             }
             if (STATE_NEQ(vertexInputParams.indexBuf))
             {
-                extGlVertexArrayElementBuffer(vao, nextState.vertexInputParams.indexBuf);
+                extGlVertexArrayElementBuffer(vao, nextState.vertexInputParams.indexBuf ? nextState.vertexInputParams.indexBuf->getOpenGLName() : 0u);
                 UPDATE_STATE(vertexInputParams.indexBuf);
             }
         }
+        if (STATE_NEQ(vertexInputParams.indirectDrawBuf))
```

Comment:
parameter buffer also needed (GL_PARAMETER_BUFFER) and a driver feature query (some externsion or GL 4.6)

Reply:
@Crisspl still need the driver feature queries for these things

Reply:
i think it's already there
`EDF_MULTI_DRAW_INDIRECT_COUNT`
https://github.com/Crisspl/IrrlichtBAW/blob/shaders/source/Irrlicht/COpenGLDriver.cpp#L1743

Reply:
resolved?

Reply:
ah indeed.